src/redux/watchList/actionCreator.js

```diff
@@ -44,3 +48,42 @@ export const replaceWatchlistStocks = (watchListId, stocks) => ({
     stocks,
   },
 });
+
+const fetchWatchlistDetailStart = () => ({
+  type: FETCH_WATCHLIST_DETAIL_START,
+});
+
+const fetchWatchlistDetailSuccess = (payload) => ({
+  type: FETCH_WATCHLIST_DETAIL_SUCCESS,
+  payload,
+});
+
+const fetchWatchlistDetailFail = (payload) => ({
+  type: FETCH_WATCHLIST_DETAIL_FAIL,
+  payload,
+});
+
+export const fetchWatchlistDetail = (symbols) => (dispatch) => {
+  const url = new URL(`${API_URL}/stock/market/batch`);
```

Comment:
doesn't it have a performance impact if the user has too many watchlists/stocks? How the batch performs? Isn't it better to fetch watchlists separately? Did you make some research on this? I would prefer the separated requests but you can convince me

Reply:
> doesn't it have a performance impact if the user has too many watchlists/stocks? How the batch performs? Isn't it better to fetch watchlists separately? Did you make some research on this? I would prefer the separated requests but you can convince me

the api can take up to 100 symbols at a time. BTW the data updates rapidly and it's the only way to keep consistency if you have the same stock in different watchlists.